### PR TITLE
tbox 1.5.2 (new formula)

### PR DESCRIPTION
**Pull request by waruqi**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/libtbox.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Libtbox < Formula
+  desc "A glib-like multi-platform c library"
+  homepage "https://tboox.org"
```
Comment (DomT4) [inline Formula/libtbox.rb]: Audit reports this isn't reachable. Perhaps not available over `https://`?

File: Formula/libtbox.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  def install
+    args = ["--smallest=y"]
+    args << "--demo=n"
+    args << "--xml=y"
+    args << "--asio=y"
+    args << "--thread=y"
+    args << "--network=y"
+    args << "--charset=y"
+
+    system "xmake", "config", *args
```
Comment (DomT4) [inline Formula/libtbox.rb]: Don't need an `args` array unless it needs manipulating. Simply:
```ruby
system "xmake", "config", "--smallest=y",
                          "--demo=n",
                          "--xml=y",
                          "--asio=y",
                          "--thread=y"
```
etc, or some reasonable combination similar to that is fine.

File: Formula/libtbox.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      {
+        if (tb_init(tb_null, tb_null))
+        {
+          tb_trace_i("hello tbox!");
+          tb_exit();
+        }
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-L#{lib}", "-ltbox", "-I#{include}", "-o", "test"
```
Comment (DomT4) [inline Formula/libtbox.rb]: We typically execute the test after creating it to ensure it works.
```ruby
system "./test"
```

File: Formula/libtbox.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class Libtbox < Formula
+  desc "A glib-like multi-platform c library"
```
Comment (DomT4) [inline Formula/libtbox.rb]: Cut the `"A"` here. Just:
```ruby
desc "Glib-like multi-platform c library"
```

**Comment by waruqi** [comment]
I have fixed these issues.  

Thansk.

**Comment by DomT4** [comment]
Just one question left: Why is this named `libtbox` rather than `tbox`? Is that something in the upstream documentation I'm missing?

**Comment by waruqi** [comment]
Oh, I thought the brew formula name of the open source library must add prefix: `lib`

It's so much the better if I can rename `libtbox` to `tbox` .

Thanks! : )

**Comment by waruqi** [comment]
I have renamed it. : )

**Comment by DomT4** [comment]
> brew formula name of the open source library must add prefix: lib

Nope! Generally speaking we follow the upstream name. If that's already taken we follow Debian. If that is already taken we'll name it something like `<project-name><developer-name>`.

Merged in https://github.com/Homebrew/homebrew-core/commit/e7e9cf626c94db9700a63cfa56d3af2f904f6993. Thank you for another contribution to Homebrew @waruqi; we appreciate it! :smiley_cat:

**Comment by waruqi** [comment]
Thanks!

**Pull request closed without merging**